### PR TITLE
Retry entries in case of service errors

### PR DESCRIPTION
**Pull request by e-ivkov**
Based on our today's discussion with @generall . We decided that we shouldn't just log and skip consensus operation entries, instead we should try to reapply them and not progress further.

### CODE REVIEW

File: lib/storage/src/content_manager/consensus_state.rs [hunk 1]
```diff
@@ -158,56 +158,87 @@ impl<C: CollectionContainer> ConsensusState<C> {
             .map_err(raft_error_other)
     }
 
-    pub fn apply_entries<T: Storage>(&self, raw_node: &mut RawNode<T>) -> Result<(), raft::Error> {
+    pub fn apply_entries<T: Storage>(&self, raw_node: &mut RawNode<T>) {
```
Comment (agourlay) [inline lib/storage/src/content_manager/consensus_state.rs:161]: Is it possible that it will retry forever and never make progress?

Comment (generall) [reply lib/storage/src/content_manager/consensus_state.rs:161]: I think we would need to do same schema with increasing timeouts. And also introduce some threshold after which we should send a proposal to consensus to disconnect the current node. But until we don;t have possibility do disconnect - yes, we need to try indefinitely

Comment (agourlay) [reply lib/storage/src/content_manager/consensus_state.rs:161]: Thanks for clarification, let's go with this solution for the time being then.

Comment (agourlay) [reply lib/storage/src/content_manager/consensus_state.rs:161]: Maybe good to create a Github issue right away about it.

**Review by agourlay — APPROVED** [review]

File: lib/storage/src/content_manager/consensus_state.rs (hunk 1 above)
Comment (e-ivkov) [reply lib/storage/src/content_manager/consensus_state.rs:161]: https://github.com/qdrant/qdrant/issues/844

**Review by generall — APPROVED** [review]

**Pull request merged**